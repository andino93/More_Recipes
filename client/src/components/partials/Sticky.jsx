import React from 'react';
import { Link } from 'react-router-dom';


const StickyComponent = ({ getAll }) => (
  <div className="container" id="sticky-anchor">
    <div className="row">
      <div className="col-lg-12">
        <ul
          className="nav justify-content-center light-well shadow-lite sticky"
          id="sticky">
          <li className="nav-item" >
            <button className="btn btn-outline-warning btn-sm" onClick={() => {
              getAll({ selected: 1 });
            }}
            style={{ marginTop: '5px' }}
            >
              <i className="fa fa-list" />
              All Recipes
            </button>
          </li>
          <li className="nav-item">
            <Link className="nav-link text-black" to="/favorites">
              <i className="fa fa-star text-warning" />
              Most Favourited
            </Link>
          </li>
        </ul>
      </div>
    </div>
  </div>
);

export default StickyComponent;
