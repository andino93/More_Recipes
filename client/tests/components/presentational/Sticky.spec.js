import React from 'react';
import expect from 'expect';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { configure } from 'enzyme';
import mockStore from '../../__mocks__/mockStore';
import mountWrap from '../../__mocks__/utils';
import Sticky from '../../../src/components/partials/Sticky.jsx';

configure({ adapter: new Adapter() });

describe('<Sticky>', () => {
  const store = mockStore({});
  const connectedSticky = mountWrap(Sticky, store, { getAll: () => {} });
  it('should render the Sticky', (done) => {
    connectedSticky.find('button').at(0).simulate('click');
    expect(toJson(connectedSticky)).toMatchSnapshot();
    done();
  });
});

