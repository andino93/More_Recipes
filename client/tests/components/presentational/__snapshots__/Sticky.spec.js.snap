// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Sticky> should render the Sticky 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "blank",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <StickyComponent
        getAll={[Function]}
      >
        <div
          className="container"
          id="sticky-anchor"
        >
          <div
            className="row"
          >
            <div
              className="col-lg-12"
            >
              <ul
                className="nav justify-content-center light-well shadow-lite sticky"
                id="sticky"
              >
                <li
                  className="nav-item"
                >
                  <button
                    className="btn btn-outline-warning btn-sm"
                    onClick={[Function]}
                    style={
                      Object {
                        "marginTop": "5px",
                      }
                    }
                  >
                    <i
                      className="fa fa-list"
                    />
                    All Recipes
                  </button>
                </li>
                <li
                  className="nav-item"
                >
                  <Link
                    className="nav-link text-black"
                    replace={false}
                    to="/favorites"
                  >
                    <a
                      className="nav-link text-black"
                      href="/favorites"
                      onClick={[Function]}
                    >
                      <i
                        className="fa fa-star text-warning"
                      />
                      Most Favourited
                    </a>
                  </Link>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </StickyComponent>
    </Router>
  </BrowserRouter>
</Provider>
`;
